test/e2e/apps/statefulset.go

```diff
@@ -850,44 +850,51 @@ var _ = SIGDescribe("StatefulSet", func() {
 		
 		It("should have a working scale subresource", func() {
 			By("Creating statefulset " + ssName + " in namespace " + ns)
-			*(ss.Spec.Replicas) = 1
+			ss := framework.NewStatefulSet(ssName, ns, headlessSvcName, 1, nil, nil, labels)
 			sst := framework.NewStatefulSetTester(c)
-			sst.PauseNewPods(ss)
-
-			_, err := c.AppsV1beta1().StatefulSets(ns).Create(ss)
+			sst.SetHttpProbe(ss)
+			ss, err := c.AppsV1beta1().StatefulSets(ns).Create(ss)
 			Expect(err).NotTo(HaveOccurred())
+			sst.WaitForRunningAndReady(*ss.Spec.Replicas, ss)
+			ss = sst.WaitForStatus(ss)
 
-			By("Saturating stateful set " + ss.Name)
-			sst.Saturate(ss)
+			By("posting a scale subresource")
+			scale := framework.NewStatefulSetScale(ss)
+			scaleResult := &appsV1beta2.Scale{}
+			err = c.AppsV1beta2().RESTClient().Put().AbsPath("/apis/apps/v1beta2").Namespace(ns).Resource("statefulsets").Name(ssName).SubResource("scale").Body(scale).Do().Into(scaleResult)
+			if err != nil {
+				framework.Failf("Failed to put scale subresource: %v", err)
+			}
+			Expect(scaleResult.Spec.Replicas).To(Equal(int32(1)))
 
 			By("getting scale subresource")
-			scale := &extensions.Scale{}
-			err = c.AppsV1beta1().RESTClient().Get().AbsPath("/apis/apps/v1beta2").Namespace(ns).Resource("statefulsets").Name(ssName).SubResource("scale").Do().Into(scale)
+			
+			err = c.AppsV1beta2().RESTClient().Get().AbsPath("/apis/apps/v1beta2").Namespace(ns).Resource("statefulsets").Name(ssName).SubResource("scale").Do().Into(scale)
```

Comment:
@liggitt The test passes on local cluster, but not on submit queue. It most probably is due to client version incompatibility. i have some ideas on resolving it.

It should not be hard moving forward - i wish to deliver gratitude for your patient help and detailed explanation. Thanks!